File: test/fixtures/java-jsoup/post_with_data_with_percent_sign.java
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import org.jsoup.Connection;
import org.jsoup.Jsoup;


class Main {

	public static void main(String[] args) throws IOException {
		Connection.Response response = Jsoup.connect("http://localhost:28139/post")
			.header("Content-Type", "application/x-www-form-urlencoded")
			.requestBody("secret=*%5*!")
			.method(org.jsoup.Connection.Method.POST)
			.ignoreContentType(true)
			.execute();

		System.out.println(response.parse());
	}
}

File: test/fixtures/javascript-jquery/post_with_data_with_percent_sign.js
$.ajax({
  url: 'http://localhost:28139/post',
  crossDomain: true,
  method: 'post',
  contentType: 'application/x-www-form-urlencoded',
  data: 'secret=*%5*!'
}).done(function(response) {
  console.log(response);
});
